wl-client/src/proxy/low_level/owned.rs

```diff
@@ -864,20 +898,33 @@ impl OwnedProxyRegistry {
 /// # Safety
 ///
 /// - `WL_INTERFACE` must be a valid wl_interface.
+/// - `mutable_type` must always return the same value.
 pub unsafe trait EventHandler {
     /// The type of interface that can be handled by this event handler.
     const WL_INTERFACE: &'static wl_interface;
 
+    /// Returns the mutable data type required by this event handler.
+    #[inline]
```

Comment:
```suggestion
    #[inline(always)]
```

iirc regular inline has issues with inlining when you declare in trait a default impl that does nothing.

Reply:
It's possible that `#[inline]` on default impls has no effect, I have not able to quickly create a test case that discriminates between it having an effect and it not having an effect.

However, I have been able to verify that default trait impls are getting inlined across crates (both trait and impl in a different crate) even without any inline annotations and even if the impl is used in the original crate. Unless the default impl has an `#[inline(never)]` annotation, the resulting binary contains 2 copies of the function body. This indeed looks like a bug because it causes code bloat, but not in a way that affects us since we always want these methods to be inlined.

Reply:
I just remember that was the case with `gdb` rust crate, where they said that unless you have `always` it sometimes doesn't work and doesn't trigger DCE, but maybe that got fixed.

Reply:
https://docs.rs/gdbstub/latest/gdbstub/target/ext/index.html#how-protocol-extensions-work---inlineable-dyn-extension-traits-idets

that's from where I remember it. It mentions that when you start nesting stuff it may stop working, so `always` is generally better for DCE to be triggered.

Reply:
I don't believe the linked documentation says anything about default trait impls in particular and dead-code elimination is not really relevant here since the code that works with this value is never specialized anyway.